Taskfile.yml

```diff
@@ -1,6 +1,12 @@
 version: "3"
 
 tasks:
+  setup:
+    - python -m venv .venv # Create a virtual environment
+    - source .venv/bin/activate # Activate the virtual environment
```

Comment:
so, i initially tried this, and unfortunately because Task runs things in subshells, the activation of the venv is borderline impossible (would require massive backflips not worth the price of admission) so i think the venv stuff should maybe come out. but otherwise, fantastic call 🙌🏻

Reply:
alternatively, if you disagree, we could leave this all in but people would just need to activate the venv again manually after this step. not sure if that's simpler or more confusing, open to thoughts/ideas.

Reply:
ahh yep removed thanks

Reply:
maybe a setuptools or poetry package manager might be worthwhile?

Reply:
i've noticed poetry adding some extra abstraction and thus confusion to people learning the python flow (from time teaching and consulting), so in the interest of prioritizing understanding over convenience i don't want to go that route BUT -- i'm wondering if maybe rather than a `setup` step in Task, there's a setup.sh script that does everything for the initial setup automatically just to accelerate people and they can come back to various things more modularly. not as nice as having it at all aligned in Task but perhaps the best compromise available. 🤔